### PR TITLE
Changed mactime bodyfile parser handle corrupt files #3430

### PR DESCRIPTION
**Pull request by joachimmetz**
Changed mactime bodyfile parser handle corrupt files #3430

### CODE REVIEW

File: plaso/parsers/mactime.py [hunk 1]
```diff
@@ -50,176 +56,257 @@ def __init__(self):
...
+      line = line_reader.readline()
+    except UnicodeDecodeError as exception:
+      raise errors.UnableToParseFile(
+          'unable to read line: {0:d} with error: {1!s}'.format(
+              line_number, exception))
+
+    while line:
+      values = line.split('|')
+      if len(values) < 11:
+        error_string = 'invalid number of values in line: {0:d}'.format(
```
Comment (Onager) [inline plaso/parsers/mactime.py:189]: Please add the number of values actually found in the line to this error message

File: plaso/parsers/mactime.py [hunk 2]
```diff
@@ -50,176 +56,257 @@ def __init__(self):
...
-    for column_name in ('atime', 'mtime', 'ctime', 'crtime'):
-      column_value = row.get(column_name, None)
-      if not column_value:
-        continue
+      else:
+        md5_value = values.pop(0)
+        if md5_value == '0':
+          md5_value = None
+        elif not self._MD5_RE.match(md5_value):
+          error_string = 'invalid MD5 value in line: {0:d}'.format(line_number)
```
Comment (Onager) [inline plaso/parsers/mactime.py:201]: Please add md5_value to this error message

File: plaso/parsers/mactime.py [hunk 3]
```diff
@@ -50,176 +56,257 @@ def __init__(self):
...
+
+        size_value = self._GetLastValueAsBase10Integer(
+            parser_mediator, values, 'size', line_number)
+        gid_value = self._GetLastValueAsBase10Integer(
+            parser_mediator, values, 'group identifier (GID)', line_number)
+        uid_value = self._GetLastValueAsBase10Integer(
+            parser_mediator, values, 'user identifier (UID)', line_number)
+
+        if uid_value is not None:
+          # Note that the user_sid value is expected to be a string or None.
```
Comment (Onager) [inline plaso/parsers/mactime.py:224]: What user_sid value? Did you mean uid? I don't understand what this comment is trying to convey, please review

File: plaso/parsers/mactime.py [hunk 4]
```diff
@@ -50,176 +56,257 @@ def __init__(self):
...
+        inode_value = values.pop(-1) or None
+        if '-' in inode_value:
+          inode_value, _, _ = inode_value.partition('-')
+
+        try:
+          inode_value = int(inode_value, 10)
+        except (TypeError, ValueError):
+          inode_value = None
+          parser_mediator.ProduceRecoveryWarning(
+              'invalid inode value in line: {0:d}'.format(line_number))
```
Comment (Onager) [inline plaso/parsers/mactime.py:238]: Please include inode_value in the warning message

**Review by Onager — CHANGES_REQUESTED** [review]

File: plaso/parsers/mactime.py (hunk 1 above)
Comment (joachimmetz) [reply plaso/parsers/mactime.py:189]: done

File: plaso/parsers/mactime.py (hunk 2 above)
Comment (joachimmetz) [reply plaso/parsers/mactime.py:201]: done

File: plaso/parsers/mactime.py (hunk 3 above)
Comment (joachimmetz) [reply plaso/parsers/mactime.py:224]: this relates to the user_sid attribute of MactimeEventData, I've changed the text a bit

File: plaso/parsers/mactime.py (hunk 4 above)
Comment (joachimmetz) [reply plaso/parsers/mactime.py:238]: done

**Review by Onager — APPROVED** [review]

**Pull request merged**